src/ccall.cpp

```diff
@@ -1271,7 +1287,7 @@ static jl_cgval_t emit_ccall(jl_codectx_t &ctx, jl_value_t **args, size_t nargs)
 #ifdef _OS_WINDOWS_
             if ((f_lib == JL_EXE_LIBNAME) || // preventing invalid pointer access
                 (f_lib == JL_LIBJULIA_INTERNAL_DL_LIBNAME) ||
-                (!strcmp(f_lib, JL_LIBJULIA_DL_LIBNAME)) ||
+                (f_lib == JL_LIBJULIA_DL_LIBNAME) ||
```

Comment:
Is this guaranteed?  What happens if someone does `ccall((foo, "libjulia"), ...)`?

Reply:
results are undefined (and always have been)

Reply:
but generally is supposed to prohibit this from triggering the optimizations here